src/distributions/base-installer.ts

```diff
@@ -129,7 +129,7 @@ export abstract class JavaBase {
       stable = false;
     }
 
-    if (!semver.validRange(version)) {
+    if (this.distribution !== 'jdkfile' && !semver.validRange(version)) {
```

Comment:
If it does not follow SemVer notation it will lead possible drawback to versions in toolcache directory because toolkit find method relies on SemVer notation.
- https://github.com/actions/toolkit/blob/main/packages/tool-cache/src/tool-cache.ts#L495